kola/tests/bpf/bpf.go

```diff
@@ -0,0 +1,117 @@
+// Copyright © Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bpf
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/pkg/capnslog"
+)
+
+var (
+	plog = capnslog.NewPackageLogger("github.com/coreos/mantle", "kola/tests/bpf")
+	// cmdPrefix is a temporary hack to pull `bcc` tools into Flatcar
+	cmdPrefix = "docker run -d --name %s -v /lib/modules:/lib/modules -v /sys/kernel/debug:/sys/kernel/debug -v /sys/fs/cgroup:/sys/fs/cgroup -v /sys/fs/bpf:/sys/fs/bpf --privileged --net host --pid host quay.io/iovisor/bcc %s"
+)
+
+// Log defines the standarg log format
+// from Docker
+// https://docs.docker.com/config/containers/logging/json-file/
+type Log struct {
+	Log    string `json:"log"`
+	Time   string `json:"time"`
+	Stream string `json:"stream"`
+}
+
+func init() {
+	register.Register(&register.Test{
+		Run:         execsnoopTest,
+		ClusterSize: 1,
+		Name:        `bpf.execsnoop`,
+		Distros:     []string{"cl"},
+		// required while SELinux policy is not correcly updated to support
+		// `bpf` and `perfmon` permission.
+		Flags: []register.Flag{register.NoEnableSelinux},
+	})
+}
+
+func execsnoopTest(c cluster.TestCluster) {
+	c.Run("filter name and args", func(c cluster.TestCluster) {
+		m := c.Machines()[0]
+		containerName := "execsnoop"
+
+		// filter commands with `docker ps`
+		plog.Infof("running %s container", containerName)
+		cmd := fmt.Sprintf(cmdPrefix, containerName, "/usr/share/bcc/tools/execsnoop -n docker -l ps")
+		if _, err := c.SSH(m, cmd); err != nil {
+			c.Fatalf("unable to run SSH command '%s': %v", cmd, err)
+		}
+
+		// sleep 2s to be sure that the container is correctly started
+		time.Sleep(2 * time.Second)
```

Comment:
Meh, I suppose that there isn't any better way here, right?

Reply:
Not a big fan neither to use `time.Sleep` - I already see the flaky tests in the CI.

Now you say it, I think we can try to use the `util.Retry`: https://github.com/flatcar-linux/mantle/blob/2ad720efa5796c258e1ddf4d5d55ddad7ddc5715/util/retry.go#L22-L25

To assert that the container is correctly started.

Reply:
That certainly would be an improvement, thanks. :)

Reply:
It's done - in the `Retry` method, we run a `docker ps` command if this one does not generate any logs, it means that `execsnoop` is not yet fully started.